geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/AvroSimpleFeatureTypeParser.scala

```diff
@@ -0,0 +1,340 @@
+/***********************************************************************
+ * Copyright (c) 2013-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.features.avro
+
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericRecord
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder
+import org.joda.time.format.ISODateTimeFormat
+import org.locationtech.geomesa.features.avro.AvroSimpleFeatureTypeParser.GeomesaAvroProperty.DeserializationException
+import org.locationtech.geomesa.utils.text.{WKBUtils, WKTUtils}
+import org.locationtech.jts.geom.{Geometry, GeometryCollection, LineString, MultiLineString, MultiPoint, MultiPolygon, Point, Polygon}
+import org.opengis.feature.simple.SimpleFeatureType
+
+import java.nio.ByteBuffer
+import java.util.{Date, Locale}
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
+import scala.reflect.ClassTag
+
+object AvroSimpleFeatureTypeParser {
+  /**
+   * Convert an Avro [[Schema]] into a [[SimpleFeatureType]].
+   */
+  def schemaToSft(schema: Schema): SimpleFeatureType = {
+    val builder = new SimpleFeatureTypeBuilder
+    builder.setName(schema.getName)
+    schema.getFields.foreach { field =>
+      val fieldName = field.name()
+      val metadata = parseMetadata(field)
+      metadata match {
+        case GeomMetadata(_, geomType, default) =>
+          builder.add(fieldName, geomType)
+          // if more than one field is set as the default geometry, the last one becomes the default
+          if (default) { builder.setDefaultGeometry(fieldName) }
+        case DateMetadata(_) =>
+          builder.add(fieldName, classOf[Date])
+        case FeatureVisibilityMetadata | NoMetadata =>
```

Comment:
I tried to emulate what was there before: https://github.com/locationtech/geomesa/blob/04ddb1626807a7f032d4f7b44cad972bd11bddf2/geomesa-kafka/geomesa-kafka-confluent/src/main/scala/org/locationtech/geomesa/kafka/confluent/ConfluentFeatureSerializer.scala#L87. How else would it get in the feature user data, should it be in the user data of the SFT?

Reply:
i'm not sure what you would prefer. if it's an attribute, it shows up when you query the layer in geoserver. if not, then it won't. either way, it only controls the visibility by being set in the user data of the feature when you create it (that's what the SecurityUtils call is doing). if you don't want it to show up in the layer, you would just set it in the user data when you read that field in the avro file instead of first setting it as an attribute, then setting the attribute as the user data.

Reply:
The schemas I'm making this for do seem to have an attribute for it

Reply:
I added another property so that schema fields can be excluded from the sft

Reply:
I added another property so you can make exclude it from the SFT